stays/views.py

```diff
@@ -78,3 +82,44 @@ def get(self, request, stay_id):
             "check_out"  : room[0].check_out
         }
         return JsonResponse(data, status=200)
+
+class StayRoomsView(View):
+    def get(self, request, stay_id):
+        if not Staytype.objects.filter(id = stay_id).exists():
+            return JsonResponse({"message":"INVALID_ID"}, status=404)
+
+        check_in , check_out = request.GET.get('CheckIn'),request.GET.get('CheckOut')
+
+        if not (check_in and check_out):
+            return JsonResponse({"message":"INVALID_DATE"}, status=400)
+
+        check_in  = datetime.strptime(check_in,"%Y-%m-%d")
+        check_out = datetime.strptime(check_out,"%Y-%m-%d")
+
+        if check_in > check_out:
+            return JsonResponse({"message":"INVALID_DATE"}, status=400)
+
+        q1 = Q(book__check_in__gte=check_in) & Q(book__check_in__lte=check_out)
+        q2 = Q(book__check_out__gte=check_in) & Q(book__check_out__lte=check_out)
+        q3 = Q(book__check_in__lt=check_in,book__check_out__gt=check_out)
+
+        rooms = Room.objects.annotate(
+            remain       = F('quantity')-Count('book__id',filter=(q1|q2|q3)),
+            is_available = Case(
+                When(
+                    remain = 0,
+                    then   = False), default = True)).filter(staytype__id=stay_id).exclude(is_available=False)
+        data = [{
+            'id'       : room.id,
+            'name'     : room.name,
+            'image_url': room.image_url,
+            'people'   : room.people,
+            'option'   : [{
+                "type"     : option.name,
+                "check_in" : option.check_in,
+                "check_out": option.check_out,
+                "price"    : discount_prices(price=option.price,check_in=check_in,check_out=check_out)
+            } for option in room.roomoption_set.all()]
```

Comment:
아래 부분에서 데이터가 많아지는 경우 호출이 잦아질 것 같습니다.
`prefetch_related()`를 사용해서 쿼리를 줄여보세요!

```suggestion
            } for option in room.roomoption_set.all()]
```